### DISCUSSION TITLE
ArrayD example for 7 or higher dimension

### DISCUSSION THREAD

**Discussion by ungsik** [rust-ndarray/ndarray#1451] [Q&A]
I created an 8 dimension ArrayD.
I can not find how to index the ArrayD for 8 dim.
Compiler raises 
the trait `NdIndex<Dim<IxDynImpl>>` is not implemented for `[usize; 8]` error when addressing
array[[i, j, k, l, m, n, o, p]].
When tuple form is tried, 
the trait `NdIndex<Dim<IxDynImpl>>` is not implemented for `(usize, usize, usize, usize, usize, usize, usize, usize)` error is raised.

**Comment by jturner314** [comment]
See the [docs for the `Index` implementation](https://docs.rs/ndarray/0.15.3/ndarray/struct.ArrayBase.html#impl-Index%3CI%3E), which indicate that the index needs to implement [`NdIndex<D>`](https://docs.rs/ndarray/0.15.3/ndarray/trait.NdIndex.html). The [`NdIndex<D>`](https://docs.rs/ndarray/0.15.3/ndarray/trait.NdIndex.html) provide a list of the implementations. The relevant ones for your case are:

- [`impl<'a> NdIndex<Dim<IxDynImpl>> for &'a [Ix]`](https://docs.rs/ndarray/0.15.3/ndarray/trait.NdIndex.html#impl-NdIndex%3CDim%3CIxDynImpl%3E%3E-for-%26%27a%20%5BIx%5D)
- [`impl<'a> NdIndex<Dim<IxDynImpl>> for &'a IxDyn`](https://docs.rs/ndarray/0.15.3/ndarray/trait.NdIndex.html#impl-NdIndex%3CDim%3CIxDynImpl%3E%3E-8)
- [`impl<D> NdIndex<D> for D where D: Dimension`](https://docs.rs/ndarray/0.15.3/ndarray/trait.NdIndex.html#impl-NdIndex%3CD%3E)

For example, you could use `array[&[i, j, k, l, m, n, o, p][..]]`.

**Comment by emmatyping** [comment]
It looks like #980 will implement for `[usize; T]` using const generics, so in 0.16.x you should be able to use `array[[i, j, k, l, m, n, o, p]]`, and the compiler error you got should go away :)
